oai_agents/common/teammates_collection.py

```diff
@@ -267,24 +267,24 @@ def generate_TC(args,
 
 
 def get_best_SP_agent(args, population):
+    all_agents = get_all_agents( args.layout_names[0], population)
     agents_scores_averaged_over_layouts = []
 
-    for layout_name in args.layout_names:
-        all_agents = [agent for agent in population[layout_name]]
-
     for agent in all_agents:
         scores = [agent.layout_scores[layout_name] for layout_name in args.layout_names]
         agents_scores_averaged_over_layouts.append((agent, sum(scores)/len(scores)))
     best_agent = max(agents_scores_averaged_over_layouts, key=lambda x: x[1])
     return best_agent[0]
 
-
+def get_all_agents(layout_name, population):
```

Comment:
recommend renaming this to `get_all_agents_for_layout()`

Reply:
Good suggestion. I will rename it.